src/PKSim.Core/Services/InteractiveSimulationRunner.cs

```diff
@@ -92,10 +109,11 @@ private async Task runSimulationAsync(Simulation simulation, bool selectOutput)
          {
             if (_cancellationTokenSources.ContainsKey(simulation))
             {
+               var end = SystemTime.UtcNow();
+               var timeSpent = end - begin;
                if (_cancellationTokenSources.TryRemove(simulation, out var ctsToDispose))
-               {
                   ctsToDispose.Dispose();
-               }
+               _executionContext.PublishEvent(new SimulationRunFinishedEvent(simulation, timeSpent));
```

Comment:
I am not sure about this. Let say it was cancelled. Is this possible that you do not throw the even then? (Becaise you only do it in the if